compiler/optimizations/optimizeOnClauses.cpp

```diff
@@ -89,12 +89,12 @@ classifyPrimitive(CallExpr *call) {
       case PRIM_BLOCK_LOCAL:
         return FAST_AND_LOCAL;
 
-      // By themselves, loops are considered "fast".
+      // Loops can have arbitrary trip counts, don't consider fast
       case PRIM_BLOCK_WHILEDO_LOOP:
       case PRIM_BLOCK_DOWHILE_LOOP:
       case PRIM_BLOCK_FOR_LOOP:
       case PRIM_BLOCK_C_FOR_LOOP:
-        return FAST_AND_LOCAL;
+        return LOCAL_NOT_FAST;
```

Comment:
I don't think these primitives actually exist at this point in compilation anymore, but https://github.com/chapel-lang/chapel/commit/8d2e84b008 was created specifically because they did exist at some point in the past so I got nervous about removing them.